src/library.tests/WorkloadInformationProviderTests.cs

```diff
@@ -1,109 +1,174 @@
+        { 
+            // The ports provided to the service
+            List<int> portsOffered = new List<int> {300, 450, 180, 190, 312};            
+            ConfigureHeadlessService(numServices: numServices, namingFunction: (i) => $"myapp-{i}", numAddresses: numAddresses, addressHostNamingFunction: (i) => $"Host-{i}", "", portsOffered);            
+            var result = await _workloadInformationProvider.GetReachableEndpointsAsync(namespaceName: "", localProcessConfig: null, includeSameNamespaceServices: true, cancellationToken: default(CancellationToken));                       
+            List<int> portsInResult = new List<int>();
+            foreach (var endpoint in result)
+            {
+                if (endpoint.Ports.Any())
+                {
+                    portsInResult.AddRange(endpoint.Ports.Select( p => p.RemotePort));
+                }
+            }
+            Assert.True(portsInResult.All(p => portsOffered.Contains(p)));
+        }
+
+        [Theory]
+        [InlineData(5, 20)]
+        public async void GetReachableServicesAsync_PortsToIgnoreIncorrectFormat_HeadlessService(int numServices, int numAddresses)
+        { 
+            ConfigureHeadlessService(numServices: numServices, namingFunction: (i) => $"myapp-{i}", numAddresses: numAddresses, addressHostNamingFunction: (i) => $"Host-{i}", "ajhfja, 19o");            
+            await Assert.ThrowsAsync<UserVisibleException>(() => _workloadInformationProvider.GetReachableEndpointsAsync(namespaceName: "", localProcessConfig: null, includeSameNamespaceServices: true, cancellationToken: default(CancellationToken)));                       
+        }
+
+        private void ConfigureHeadlessService(int numServices, Func<int, string> namingFunction, int numAddresses, Func<int, string> addressHostNamingFunction, string ignorePorts = null, List<int> ports = null)
+        {
+            var serviceList = new List<V1Service>();
+            // introducing this variable so we can add an endpoint with empty subset to have crash coverage
+            bool addSubset = false;
+            for (int i = 0; i < numServices; i++)
+            {
+                var service = new V1Service()
+                {
+                    Spec = new V1ServiceSpec()
+                    {
+                        Type = "ClusterIP",
+                        ClusterIP = "None",
+                        Ports = new List<V1ServicePort> { new V1ServicePort(port: ports?.ElementAtOrDefault(i) ?? 80, protocol: "TCP") },
+                        Selector = new Dictionary<string, string> { { "app", "myapp" } }
+                    },
+                    Metadata = new V1ObjectMeta()
+                    {
+                        Name = namingFunction(i),
+                    }
+                };
+
+                if (!string.IsNullOrEmpty(ignorePorts))
+                {
+                    service.Metadata.Annotations = new Dictionary<string, string>();
```

Comment:
this add all same ports to ignore for every service. My guess is that user would configure different ports to ignore for different services.